internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -306,3 +340,46 @@ func (q amsQuotaService) DeleteQuota(subscriptionId string) *errors.ServiceError
 	}
 	return nil
 }
+
+func (q amsQuotaService) DeleteQuotaForBillingModel(subscriptionId string, kafkaBillingModel config.KafkaBillingModel) *errors.ServiceError {
+	if subscriptionId == "" {
+		return nil
+	}
+
+	subscription, found, err := q.GetSubscriptionByID(subscriptionId)
+	if err != nil {
+		return errors.GeneralError("failed to delete the quota: %v", err)
+	}
+
+	if !found {
+		return nil
+	}
+
+	if subscription.Plan().ID() == kafkaBillingModel.AMSProduct {
+		rr, err := q.amsClient.GetReservedResourcesBySubscriptionID(subscriptionId)
+		if err != nil {
+			return errors.GeneralError("failed to delete the quota: %v", err)
+		}
+		if arrays.AnyMatch(rr, func(resource *amsv1.ReservedResource) bool {
+			return resource.ResourceName() == kafkaBillingModel.AMSResource
+		}) {
+			// this subscription is related to this billing model. We can delete it.
+			return q.DeleteQuota(subscriptionId)
+		}
+	}
+
+	// the specified subscription id is not related to the specified billing model
+	return nil
+}
+
+func (q amsQuotaService) GetSubscriptionByID(subscriptionId string) (*amsv1.Subscription, bool, *errors.ServiceError) {
```

Comment:
Let's document the behavior of this method in the different cases: is not finding it an error? a non-error but with the subscription nil? etc...